streams/src/test/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessorMetricsTest.java

```diff
@@ -0,0 +1,219 @@
+            mkEntry("buffer-id", "test-store")
+        )
+    );
+
+    private static final MetricName BUFFER_COUNT_AVG_METRIC = new MetricName(
+        "suppression-buffer-count-avg",
+        "stream-buffer-metrics",
+        "The average count of buffered records.",
+        mkMap(
+            mkEntry("client-id", "mock-processor-context-virtual-thread"),
+            mkEntry("task-id", "0_0"),
+            mkEntry("buffer-id", "test-store")
+        )
+    );
+
+    private static final MetricName BUFFER_COUNT_CURRENT_METRIC = new MetricName(
+        "suppression-buffer-count-current",
+        "stream-buffer-metrics",
+        "The current count of buffered records.",
+        mkMap(
+            mkEntry("client-id", "mock-processor-context-virtual-thread"),
+            mkEntry("task-id", "0_0"),
+            mkEntry("buffer-id", "test-store")
+        )
+    );
+
+    private static final MetricName BUFFER_COUNT_MAX_METRIC = new MetricName(
+        "suppression-buffer-count-max",
+        "stream-buffer-metrics",
+        "The max count of buffered records.",
+        mkMap(
+            mkEntry("client-id", "mock-processor-context-virtual-thread"),
+            mkEntry("task-id", "0_0"),
+            mkEntry("buffer-id", "test-store")
+        )
+    );
+
+    @Test
+    public void shouldRecordMetrics() {
+        final String storeName = "test-store";
+
+        final StateStore buffer = new InMemoryTimeOrderedKeyValueBuffer.Builder(storeName)
+            .withLoggingDisabled()
+            .build();
+
+        final KTableSuppressProcessor<String, Long> processor =
+            new KTableSuppressProcessor<>(
+                (SuppressedInternal<String>) Suppressed.<String>untilTimeLimit(Duration.ofDays(100), maxRecords(1)),
+                storeName,
+                String(),
+                new FullChangeSerde<>(Long())
+            );
+
+        final MockInternalProcessorContext context = new MockInternalProcessorContext();
+        context.setCurrentNode(new ProcessorNode("testNode"));
+
+        buffer.init(context, buffer);
+        processor.init(context);
+
+        final long timestamp = 100L;
+        context.setStreamTime(timestamp);
+        context.setRecordMetadata("", 0, 0L, null, timestamp);
+        final String key = "longKey";
+        final Change<Long> value = new Change<>(null, ARBITRARY_LONG);
+        processor.process(key, value);
+
+        {
+            final Map<MetricName, ? extends Metric> metrics = context.metrics().metrics();
+
+            assertThat(metrics.get(EVICTION_RATE_METRIC).metricValue(), is(0.0));
+            assertThat(metrics.get(EVICTION_TOTAL_METRIC).metricValue(), is(0.0));
+            assertThat((Double) metrics.get(SUPPRESSION_RATE_METRIC).metricValue(), greaterThan(0.0));
+            assertThat(metrics.get(SUPPRESSION_TOTAL_METRIC).metricValue(), is(1.0));
+            assertThat(metrics.get(BUFFER_SIZE_AVG_METRIC).metricValue(), is(51.0));
```

Comment:
I feel like I should know the answer to this, but how do we `51` for the answer here and below?

Reply:
Empirically :/

It's just the result of summing all the pieces of the test data.